components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OASParserUtil.java

```diff
@@ -655,6 +652,69 @@ private static String getComponentCategory(String ref) {
         return "";
     }
 
+    /**
+     * Extract the archive file and validates the openAPI definition
+     *
+     * @param inputStream   file as input stream
+     * @param returnContent whether to return the content of the definition in the response DTO
+     * @return APIDefinitionValidationResponse
+     * @throws APIManagementException if error occurred while parsing definition
+     */
+    public static APIDefinitionValidationResponse extractAndValidateOpenAPIArchive(InputStream inputStream, boolean returnContent)
+            throws APIManagementException, IOException {
+        String path = System.getProperty(APIConstants.JAVA_IO_TMPDIR) + File.separator + APIConstants.OPENAPI_ARCHIVES_TEMP_FOLDER + File.separator + UUID
+                .randomUUID().toString();
+        String archivePath = path + File.separator + APIConstants.OPENAPI_ARCHIVE_ZIP_FILE;
+        String extractedLocation = APIFileUtil
+                .extractUploadedArchive(inputStream, APIConstants.OPENAPI_EXTRACTED_DIRECTORY, archivePath, path);
+        File[] listOfFiles = new File(extractedLocation).listFiles();
+        File archive_directory = null;
+        if (listOfFiles != null) {
+            for (File file: listOfFiles) {
+                if (file.isDirectory()) {
+                    archive_directory = file.getAbsoluteFile();
+                }
+            }
+        }
+        if (archive_directory == null) {
+            log.error("Could not find an archive in the given ZIP file ");
+            throw new FileNotFoundException();
+        }
+        File master_swagger;
+        if ((new File(archive_directory + APIConstants.OPENAPI_MASTER_JSON)).exists()) {
+            master_swagger = new File(archive_directory + APIConstants.OPENAPI_MASTER_JSON);
```

Comment:
variable names should be in camelCase ie: archive_directory, master_swagger

Reply:
https://github.com/wso2/carbon-apimgt/commit/bbdcfe5dd9f994b5693c99f127caa4948e5f3f87